lndr.module

```diff
@@ -366,166 +310,129 @@ function lndr_sync_path() {
   foreach ($projects as $project) {
-    if (strstr($project['publish_url'], $base_url)) {
-      $drupal_pages[] = $project;
-    }
+    $new_projects[$project["id"]] = $project;
   }
-  // Nothing to process
-  if (empty($drupal_pages)) {
+
+  if (empty($new_projects)) {
     return;
   }
-  // @todo: We are ignoring status overall right now in general, should we sync this with Lndr?
-  // @todo: However this is also something that needs to go into a workbench flow.
-  // Going through all the pages that are published to this URL
-  foreach ($drupal_pages as $page) {
-    $path = substr($page['publish_url'], strlen($base_url));
-    $path = ltrim($path, '/');
-
-    // Looking to see if a system path exist based on the alias given
-    $existing_system_path_by_alias = drupal_lookup_path('source', $path);
-    if ($existing_system_path_by_alias) {
-      // Let's see if we can find a corresponding node with the system path found
-      // The system path should be node/[nid]
-      $nid = explode('/', $existing_system_path_by_alias);
-      $nid = $nid[1];
-      $node = node_load($nid);
-      if (!empty($node)) {
-        // case 1. this node was reserved for this lndr page, we add a node data to sync with
-        // lndr project, as well as publish it.
-        $lang = $node->language;
-        if (empty($node->lndr_project_id)) {
-          // $node->status = 1;
-          $node->lndr_project_id[$lang][0]['value'] = $page['id'];
-          node_save($node);
-        }
-      }
-    } else {
-      // case 3. let's see if a previous node is stored, but we updated to a new path from Lndr
-      // within the same domain
-      $query = new EntityFieldQuery();
-      $query->entityCondition('entity_type', 'node')
-        ->entityCondition('bundle', LNDR_CT_TYPE)
-        ->fieldCondition('lndr_project_id', 'value', $page['id'], '=');
-      $result = $query->execute();
-      if (isset($result['node'])) {
-        // Making sure that it is still on the same domain
-        if (substr($page['publish_url'], 0, strlen($base_url)) === $base_url) {
-          // Extracting the path
-          $_path = ltrim(substr($page['publish_url'], strlen($base_url)), '/');
-          // Compare that path with the drupal node alias
-          $result_node = current($result['node']);
-          $node_alias = drupal_lookup_path('alias', 'node/' . $result_node->nid);
-          if ($_path !== $node_alias) {
-            // Update that node alias
-            $node_to_be_updated = node_load($result_node->nid);
-            $node_to_be_updated->path['alias'] = $_path;
-            node_save($node_to_be_updated);
+  $empty_projects_node = array();
+  $nodes = _lndr_load_nodes();
+  foreach ($nodes as $node) {
+    $lndr_project_id = field_get_items('node', $node, 'lndr_project_id');
+    if (empty($lndr_project_id)) {
+      $empty_projects_node[] = $node->nid;
+    }
+    else {
+      $lndr_id = (reset($lndr_project_id))['value'];
```

Comment:
`$lndr_id = (reset($lndr_project_id))['value'];`
This shorthand appears as syntax error on my IDE (PhpStorm), can you please ensure PHP version compitability?
![screen shot 2018-08-15 at 3 40 55 pm](https://user-images.githubusercontent.com/253281/44169193-a8fafd00-a0a1-11e8-9840-4ac3d2dffae2.png)

Reply:
This syntax is allowed since PHP 5.4. In my opinion, we should not support "dinosaurs" like as PHP 5.3, because the latest release was 4 years ago :)